### ISSUE TITLE
ECE button load tracks events not triggered when more than 1 button is loaded on a page

### ISSUE THREAD

**Issue by reykjalin** [Automattic/woocommerce-payments#9795]
> […] Apple Pay button load events aren't being triggered because of how triggering the tracks event is debounced.
>
> The `onReadyHandler` is triggered separately for the APay and GPay buttons so `trackExpressCheckoutButtonLoad` is called twice _with different arguments each time_. This means the second call is debounced and not sent.
>
> _Originally posted by @reykjalin in https://github.com/Automattic/woocommerce-payments/pull/9793#pullrequestreview-2455811995_

See usage:

https://github.com/Automattic/woocommerce-payments/blob/8b1bcda6da35a129f66b975909c7a3ebb3ed219f/client/express-checkout/event-handlers.js#L136-L139

https://github.com/Automattic/woocommerce-payments/blob/8b1bcda6da35a129f66b975909c7a3ebb3ed219f/client/express-checkout/tracking.js#L21-L36

## Proposed solution

The straight-forward fix is to split the tracks events into 2 debounced functions: 1 for Apple Pay and another for Google Pay.

We might still be able to abstract this away by calling those 2 separately debounced functions from this interface, and that way we won't have to make any changes to how the tracks events are triggered throughout the codebase.

**Comment by pierorocca** [comment]
@malithsen correct me if I'm wrong, we decided we'd remove this low value tracks event to lessen the server burden?

**Comment by malithsen** [comment]
>  correct me if I'm wrong, we decided we'd remove this low value tracks event to lessen the server burden?

Yes, we discussed about removing `_load` events along with https://github.com/Automattic/woocommerce-payments/issues/9677

Admittedly, I forgot at the time that @rafaelzaleski might be relying on them for express checkout reports he worked on. 

@rafaelzaleski Are we currently using these `_load` events or plan to use them in the future for any of the Looker reports you worked on? If not, I think we should just remove them.

**Comment by rafaelzaleski** [comment]
We're currently using button loads in the ECE Looker dashboard for:

- Post-Release Monitoring: After a new release, a sudden drop in load events could signal that Express Checkout isn't working properly.
- Debugging Click Issues: By looking at both load and click events, we can see if buttons are loading but not being clicked. For instance, fewer clicks without fewer loads might mean the buttons are there but not clickable.
- Could potentially help us see how payment methods are converting button loads into successful checkouts by tracking both loads and clicks.

@pierorocca @malithsen, when you mention "server burden," are you talking about the **merchant server** or the **Tracks server**? Has this been flagged anywhere, or is it causing any specific issues that we need to address, such as slowing down Looker dashboards?

The `_load` event isn't critical—it's just nice to have. If it's not worth the cost, we can remove it. But if it's causing server issues, we can look into optimizing it so we don't lose these insights.

cc @bborman22

**Comment by malithsen** [comment]
> when you mention "server burden," are you talking about the merchant server or the Tracks server? 

It's the burden on the merchant's server. Currently, these events are recorded through an Ajax event from the shopper's browser to the merchant server, which then emits a request to the Tracks backend. Each event creates a new Ajax request, a concern was raised in p1730725717671609-slack-C7U3Y3VMY about the large number of these requests on high-traffic merchants.  

We were thinking of removing `_load` events as it looked like it wasn't being used anywhere in Looker. Since there's a legitimate use case for these events, we can keep them. 
The other option discussed in relation to this issue is making shopper events follow the merchant's Tracks opt-out setting. This way, merchants have control over the requests that hit their servers.

**Comment by rafaelzaleski** [comment]
@malithsen, does using `window.wcTracks.recordEvent` to record the event also send data to the merchant backend? Would that address the issue with excessive ajax calls?

**Comment by malithsen** [comment]
@rafaelzaleski `window.wcTracks.recordEvent` IIRC, this is the WC core's Tracks integration. It has a couple of limitations that prevent us from using it for shopper events.

- `wcadmin` prefix is hard-coded to the library. We want a separate prefix (`wcpay`) for shopper events.
- Tracks scripts are only loaded in the admin area.

That said, Tracks does have a JS library that we can use to record directly from the browser without hitting the merchant server. The downside is that events can be blocked by adblockers running on the shopper's browser.

**Comment by bborman22** [comment]
I saw this issue got closed from the attached PR, but was wondering if we need to capture the merchant server load concern in another issue so it doesn't get lost. I could see it being addressed through an alternative tracking method or by removing things like the `_load` events is that accurate?

**Comment by rafaelzaleski** [comment]
> I could see it being addressed through an alternative tracking method or by removing things like the _load events is that accurate?

Yes, that's correct. If we don’t need the numbers to be completely accurate, we can trigger the event from the frontend directly to the Tracks server. However, users with ad blockers won’t be tracked.
